feat(baseline): Validate mutual exclusivity of explicit --since and --with-baseline options (#2723)
stryker-mutator/stryker-net: docs/configuration.md

Set the diffing target on the command line by passing a committish with the since flag.
Set the diffing target in the config file by setting the [since target](#sincetarget-committish) option.

> *\* The with-baseline and since options are mutually exclusive. This feature implicitly enables the [since](#since-flag-committish) feature.*

Reviewer: ```suggestion
*\* The baseline and since features are mutually exclusive. This feature implicitly enables the [since](#since-flag-committish) feature for now.*
```
